web3_proxy/src/frontend/users.rs

```diff
@@ -181,25 +177,28 @@ pub async fn user_login_post(
     // we can't trust that they didn't tamper with the message in some way. like some clients return it hex encoded
     // TODO: checking 0x seems fragile, but I think it will be fine. siwe message text shouldn't ever start with 0x
     let their_msg: Message = if payload.msg.starts_with("0x") {
-        let their_msg_bytes = Bytes::from_str(&payload.msg).context("parsing payload message")?;
+        let their_msg_bytes =
+            Bytes::from_str(&payload.msg).web3_context("parsing payload message")?;
 
         // TODO: lossy or no?
         String::from_utf8_lossy(their_msg_bytes.as_ref())
             .parse::<siwe::Message>()
-            .context("parsing hex string message")?
+            .web3_context("parsing hex string message")?
```

Comment:
Maybe it makes sense to create a new error class for anything related to wallet verification & signing, otherwise we will get a bunch of random 5xx again?